xds/internal/resolver/xds_resolver_test.go

```diff
@@ -1459,557 +816,655 @@ func (s) TestResolverMaxStreamDuration(t *testing.T) {
 				},
 			},
-			selectErr: errUnsupportedClientRouteAction.Error(),
+			wantErr: "matched route does not have a supported route action type",
 		},
 		{
 			name: "route type RouteActionNonForwardingAction invalid for client",
-			ldsFilters: []xdsresource.HTTPFilter{
-				{Name: "foo", Filter: &filterBuilder{path: &path}, Config: filterCfg{s: "foo1"}},
-			},
-			rtCfgUpdate: xdsresource.RouteConfigUpdate{
-				VirtualHosts: []*xdsresource.VirtualHost{
-					{
-						Domains: []string{targetStr},
-						Routes: []*xdsresource.Route{{
-							Prefix: newStringP("1"),
-							WeightedClusters: map[string]xdsresource.WeightedCluster{
-								"A": {Weight: 1},
-								"B": {Weight: 1},
-							},
-							ActionType: xdsresource.RouteActionNonForwardingAction,
-						}},
-					},
+			listener: &v3listenerpb.Listener{
+				Name: defaultTestServiceName,
+				ApiListener: &v3listenerpb.ApiListener{
+					ApiListener: testutils.MarshalAny(t, &v3httppb.HttpConnectionManager{
+						RouteSpecifier: &v3httppb.HttpConnectionManager_RouteConfig{
+							RouteConfig: &v3routepb.RouteConfiguration{
+								Name: defaultTestRouteConfigName,
+								VirtualHosts: []*v3routepb.VirtualHost{{
+									Domains: []string{defaultTestServiceName},
+									Routes: []*v3routepb.Route{{
+										Match: &v3routepb.RouteMatch{
+											PathSpecifier: &v3routepb.RouteMatch_Prefix{Prefix: methodName},
+										},
+										Action: &v3routepb.Route_NonForwardingAction{},
+									}},
+								}},
+							}},
+						HttpFilters: []*v3httppb.HttpFilter{e2e.RouterHTTPFilter},
+					}),
 				},
 			},
-			rpcRes: map[string][][]string{
-				"1": {
-					{"build:foo1", "override:foo2", "build:bar1", "override:bar2", "newstream:foo1", "newstream:bar1", "done:bar1", "done:foo1"},
+			wantErr: "matched route does not have a supported route action type",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			// Spin up an xDS management server.
+			ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+			defer cancel()
+			nodeID := uuid.New().String()
+			mgmtServer, _, _ := setupManagementServerForTest(ctx, t, nodeID)
+
+			// Build an xDS resolver.
+			stateCh, _, _ := buildResolverForTarget(t, resolver.Target{URL: *testutils.MustParseURL("xds:///" + defaultTestServiceName)})
+
+			// Update the management server with a listener resource that
+			// contains inline route configuration.
+			configureResourcesOnManagementServer(ctx, t, mgmtServer, nodeID, []*v3listenerpb.Listener{test.listener}, nil)
+
+			// Ensure that the resolver pushes a state update to the channel.
+			cs := verifyUpdateFromResolver(ctx, t, stateCh, "")
+
+			// Ensure that it returns the expected error.
+			_, err := cs.SelectConfig(iresolver.RPCInfo{Method: methodName, Context: ctx})
+			if err == nil || !strings.Contains(err.Error(), test.wantErr) {
+				t.Errorf("SelectConfig(_) = _, %v; want _, Contains(%v)", err, test.wantErr)
+			}
```

Comment:
Optional: reduce variable scope and lines: make this one line in the if conditional. The lines are long though.

Reply:
Yeah, I used to write it as one line earlier. But over time, I have seen this more in our code base and this reads better for sure. If we had a single condition in the `if` statement, it makes total sense to fold that into the assignment. But here, the conditional is more involved and it reads better when split up.

Reply:
Fair.